Server pytests are failing unexpected on unrelated changes

See http://jenkins.perf.lab.eng.bos.redhat.com/job/pbench-ci-multi/job/PR-2005/93/console

```
============================= test session starts ==============================
platform linux -- Python 3.8.6, pytest-4.6.11, py-1.8.2, pluggy-0.13.1
cachedir: /var/tmp/tox/jenkins-pytests/.pytest_cache
rootdir: /home/pbench, inifile: pytest.ini
plugins: requests-mock-1.8.0, cov-2.10.1, mock-1.10.4, helpers-namespace-2019.1.8
collecting ... 
collecting 41 items                                                            
collecting 79 items                                                            
collected 83 items                                                             

lib/pbench/test/unit/common/test_config.py ....
lib/pbench/test/unit/common/test_logging.py ......
lib/pbench/test/unit/common/test_utils.py .
lib/pbench/test/unit/agent/test_config.py ....
lib/pbench/test/unit/agent/task/test_copy_result_tb.py ....
lib/pbench/test/unit/agent/task/test_make_result_tb.py .......
lib/pbench/test/unit/agent/task/test_move_results.py .
lib/pbench/test/unit/server/test_query_controller.py ........F.....
lib/pbench/test/unit/server/test_query_month_indices.py ......
lib/pbench/test/unit/server/test_requests.py ............
lib/pbench/test/unit/server/test_utils.py .
lib/pbench/test/functional/test_pbench_config.py ...
lib/pbench/test/functional/agent/cli/commands/test_cleanup.py ..
lib/pbench/test/functional/agent/cli/commands/results/test_clear_results.py ..
lib/pbench/test/functional/agent/cli/commands/tools/test_clear_tools.py ........
lib/pbench/test/functional/agent/cli/commands/tools/test_tools_list.py ..
lib/pbench/test/functional/agent/cli/commands/triggers/test_list_triggers.py ..
lib/pbench/test/functional/agent/cli/commands/triggers/test_register_tool_trigger.py ....

=================================== FAILURES ===================================
_______________________ TestQueryControllers.test_query ________________________

self = <pbench.test.unit.server.test_query_controller.TestQueryControllers object at 0x7f955b843310>
client = <FlaskClient <Flask 'api-server'>>
server_config = <pbench.server.PbenchServerConfig object at 0x7f955b843340>
query_helper = <function query_helper.<locals>.query_helper at 0x7f955b91d790>

    def test_query(self, client, server_config, query_helper):
        """
        test_query Check the construction of Elasticsearch query URI
        and filtering of the response body.
        """
        json = {
            "user": "drb",
            "start": "2020-08",
            "end": "2020-10",
        }
        response_payload = {
            "took": 1,
            "timed_out": False,
            "_shards": {"total": 1, "successful": 1, "skipped": 0, "failed": 0},
            "hits": {
                "total": {"value": 2, "relation": "eq"},
                "max_score": None,
                "hits": [],
            },
            "aggregations": {
                "controllers": {
                    "doc_count_error_upper_bound": 0,
                    "sum_other_doc_count": 0,
                    "buckets": [
                        {
                            "key": "unittest-controller1",
                            "doc_count": 2,
                            "runs": {
                                "value": 1.59847315581e12,
                                "value_as_string": "2020-08-26T20:19:15.810Z",
                            },
                        },
                        {
                            "key": "unittest-controller2",
                            "doc_count": 1,
                            "runs": {
                                "value": 1.6,
                                "value_as_string": "2020-09-26T20:19:15.810Z",
                            },
                        },
                    ],
                }
            },
        }
    
        index = self.build_index(server_config, ("2020-08", "2020-09", "2020-10"))
>       response = query_helper(json, index, 200, server_config, json=response_payload)

lib/pbench/test/unit/server/test_query_controller.py:161: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

payload = {'end': '2020-10', 'start': '2020-08', 'user': 'drb'}
expected_index = '/unit-test.v6.run-data.2020-08,unit-test.v6.run-data.2020-09,unit-test.v6.run-data.2020-10,'
expected_status = 200
server_config = <pbench.server.PbenchServerConfig object at 0x7f955b843340>
kwargs = {'json': {'_shards': {'failed': 0, 'skipped': 0, 'successful': 1, 'total': 1}, 'aggregations': {'controllers': {'bucke...ount': 0}}, 'hits': {'hits': [], 'max_score': None, 'total': {'relation': 'eq', 'value': 2}}, 'timed_out': False, ...}}
es_url = 'http://elasticsearch.example.com:7080'
response = <Response streamed [200 OK]>
@py_assert1 = <requests_mock.request._RequestObjectProxy object at 0x7f955b866100>
@py_assert3 = 'http://elasticsearch.example.com:7080/unit-test.v6.run-data.2020-08,unit-test.v6.run-data.2020-10,/_search?ignore_unavailable=true'
@py_assert8 = 'http://elasticsearch.example.com:7080/unit-test.v6.run-data.2020-08,unit-test.v6.run-data.2020-09,unit-test.v6.run-data.2020-10,'
@py_assert9 = '/_search?ignore_unavailable=true'
@py_assert11 = 'http://elasticsearch.example.com:7080/unit-test.v6.run-data.2020-08,unit-test.v6.run-data.2020-09,unit-test.v6.run-data.2020-10,/_search?ignore_unavailable=true'

    def query_helper(payload, expected_index, expected_status, server_config, **kwargs):
        es_url = get_es_url(server_config)
        requests_mock.post(re.compile(f"{es_url}"), **kwargs)
        response = client.post(
            f"{server_config.rest_uri}/controllers/list", json=payload
        )
>       assert requests_mock.last_request.url == (
            es_url + expected_index + "/_search?ignore_unavailable=true"
        )
E       AssertionError: assert 'http://elast...vailable=true' == 'http://elasti...vailable=true'
E         Skipping 85 identical leading characters in diff, use -v to show
E         - data.2020-10,/_search?ignore_unavailable=true
E         + data.2020-09,unit-test.v6.run-data.2020-10,/_search?ignore_unavailable=true
E         ? ++++++++++++++++++++++++++++++

lib/pbench/test/unit/server/test_query_controller.py:32: AssertionError
------------------------------ Captured log setup ------------------------------
INFO     pbench.server.api:__init__.py:28 Registering service endpoints with base URI /api/v1
------------------------------ Captured log call -------------------------------
INFO     pbench.server.api:query_controllers.py:174 QueryControllers POST for user drb, prefix unit-test, session None: (2020-08-31 00:00:00 - 2020-10-31 00:00:00)
DEBUG    requests_mock.adapter:adapter.py:255 POST http://elasticsearch.example.com:7080/unit-test.v6.run-data.2020-08,unit-test.v6.run-data.2020-10,/_search?ignore_unavailable=true 200
INFO     pbench.server.api:query_controllers.py:253 2 controllers found

----------- coverage: platform linux, python 3.8.6-final-0 -----------
Coverage XML written to file /tmp/pbench/cov/cov.xml

===================== 1 failed, 82 passed in 18.56 seconds =====================
FAILED: pytest ...
ERROR: InvocationError for command /usr/bin/bash -c './jenkins/run-pytests /var/tmp/tox/jenkins-pytests' (exited with code 1)
```

--- a/lib/pbench/server/api/resources/query_apis/query_controllers.py
+++ b/lib/pbench/server/api/resources/query_apis/query_controllers.py
@@ -160,8 +160,8 @@ class QueryControllers(Resource):
                 abort(401, message="invalid user authorization")
 
         try:
-            start = parser.parse(start_arg)
-            end = parser.parse(end_arg)
+            start = parser.parse(start_arg).replace(day=1)
+            end = parser.parse(end_arg).replace(day=1)
         except Exception as e:
             self.logger.info(
                 "Invalid start or end time string: {}, {}: {}", start_arg, end_arg, e

